adapters/repos/db/vector/hnsw/distancer/cosine_dist.go

```diff
@@ -53,3 +53,11 @@ func (d CosineDistanceProvider) Type() string {
 func (d CosineDistanceProvider) New(a []float32) Distancer {
 	return &CosineDistance{a: a}
 }
+
+func (d CosineDistanceProvider) Step(x, y float32) float32 {
+	return dotProductStepImplementation(x, y)
+}
+
+func (d CosineDistanceProvider) Wrap(x float32) float32 {
+	return 1 - x
```

Comment:
I think there is no need to incure a function call we can just use 1 - x instead. It is much simpler and more readable